nix/cabal-project.nix

```diff
@@ -0,0 +1,124 @@
+    plutus-benchmark.package.buildable = !isCrossCompiling;
+    plutus-tx-plugin.package.buildable = !isCrossCompiling;
+    # Needs agda
+    plutus-metatheory.package.buildable = !isCrossCompiling;
+    # These need R
+    plutus-core.components.benchmarks.cost-model-test.buildable = l.mkForce (!isCrossCompiling);
+    plutus-core.components.exes.generate-cost-model.buildable = l.mkForce (!isCrossCompiling);
+    # This contains support for doing testing, so we're not interested in cross-compiling it
+    plutus-conformance.package.buildable = !isCrossCompiling;
+
+
+    # Packages we just don't want docs for
+    plutus-benchmark.doHaddock = false;
+
+    # FIXME: Haddock mysteriously gives a spurious missing-home-modules warning
+    plutus-tx-plugin.doHaddock = false;
+
+    # In this case we can just propagate the native dependencies for the build of
+    # the test executable, which are actually set up right (we have a
+    # build-tool-depends on the executable we need)
+    # I'm slightly surprised this works, hooray for laziness!
+    plutus-metatheory.components.tests.test1.preCheck =
+      let
+        cmp = config.hsPkgs.plutus-metatheory.components.tests.test1;
+        deps = cmp.executableToolDepends;
+      in
+      ''PATH=${l.makeBinPath deps }:$PATH'';
+
+    # FIXME: Somehow this is broken even with setting the path up as above
+    plutus-metatheory.components.tests.test2.doCheck = false;
+
+    # plutus-metatheory needs agda with the stdlib around for the custom setup
+    # I can't figure out a way to apply this as a blanket change for all the
+    # components in the package, oh well
+    plutus-metatheory.components.library.build-tools = [ nix.plutus.agda-with-stdlib ];
+    plutus-metatheory.components.exes.plc-agda.build-tools = [ nix.plutus.agda-with-stdlib ]; # editorconfig-checker-disable-line
+    plutus-metatheory.components.tests.test1.build-tools = [ nix.plutus.agda-with-stdlib ]; # editorconfig-checker-disable-line
+    plutus-metatheory.components.tests.test2.build-tools = [ nix.plutus.agda-with-stdlib ]; # editorconfig-checker-disable-line
+    plutus-metatheory.components.tests.test3.build-tools = [ nix.plutus.agda-with-stdlib ]; # editorconfig-checker-disable-line
+
+    plutus-core.components.benchmarks.update-cost-model = {
+      build-tools = [ nix.plutus.r-with-packages ];
+    };
+
+    plutus-core.components.benchmarks.cost-model-test = {
+      build-tools = [ nix.plutus.r-with-packages ];
+    };
+
+    # -Werror for CI
+    # Only enable on the newer compilers. We don't care about warnings on the old ones,
+    # and sometimes it's hard to be warning free on all compilers, e.g. the unused
+    # packages warning is bad in 8.10.7
+    # (https://gitlab.haskell.org/ghc/ghc/-/merge_requests/6130)
+
+    # Werror everything.
+    # This is a pain, see https://github.com/input-output-hk/haskell.nix/issues/519
+
+    plutus-benchmark.ghcOptions = l.mkIf isNotGhc8107 [ "-Werror" ];
```

Comment:
This seems worse than the previous structure that just had a single if for the whole block of things... Same with the cross-compiling block. I don't think that flattening all the conditionals like this is actually helpful?